Application/EdFi.Security.DataAccess/Repositories/ISecurityRepository.cs

```diff
@@ -4,35 +4,39 @@
 // See the LICENSE and NOTICES files in the project root for more information.
 
 using System.Collections.Generic;
+using Autofac.Extras.DynamicProxy;
+using Castle.DynamicProxy;
 using EdFi.Security.DataAccess.Models;
 
-namespace EdFi.Security.DataAccess.Repositories
-{
-    public interface ISecurityRepository
-    {
-        Action GetActionByHttpVerb(string httpVerb);
+namespace EdFi.Security.DataAccess.Repositories;
 
-        Action GetActionByName(string actionName);
+/// <summary>
+/// Defines higher-level methods for obtaining security metadata for making authorization decisions.
+/// </summary>
+/// <remarks>Implementations of this interface must be configured with a named <see cref="IInterceptor" /> registration of "cache-security".</remarks>
+[Intercept("cache-security")]
```

Comment:
I'm not sure what this `[Intercept(...)]` does. Are you sure it is necessary for the .NET 8 upgrade?